File: test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/FunctionApps/dotnet-isolated/EventHub/MultiKafkaTriggerQueueOutput.cs
using System;
using Microsoft.Azure.Functions.Worker;
using Microsoft.Extensions.Logging;
using Microsoft.Azure.Functions.Worker.Http;
using Newtonsoft.Json.Linq;


namespace EventHub
{
    public class MultiKafkaTriggerQueueOutput
    {
        // KafkaTrigger sample 
        // Consume the message from "topic" on the LocalBroker.
        // Add `EventHubBrokerList` and `EventHubConnectionString` to the local.settings.json
        // For EventHubs
        // "EventHubBrokerList": "{EVENT_HUBS_NAMESPACE}.servicebus.windows.net:9093"
        // "EventHubConnectionString":"{EVENT_HUBS_CONNECTION_STRING}
        [Function("MultiKafkaTriggerQueueOutput")]
        [QueueOutput("e2e-dotnet-isolated-multi-eventhub")]
        public static string[] Run(
            [KafkaTrigger("EventHubBrokerList",
                          "e2e-kafka-dotnet-isolated-multi-eventhub",
                          Username = "$ConnectionString",
                          Password = "%EventHubConnectionString%",
                          Protocol = BrokerProtocol.SaslSsl,
                          AuthenticationMode = BrokerAuthenticationMode.Plain,
                          ConsumerGroup = "$Default",
                          IsBatched = true)] string[] events, FunctionContext context)
        {
            string[] messages = new string[events.Length];
            for (int i = 0; i < events.Length; i++)
            {
                var logger = context.GetLogger("KafkaFunction");
                logger.LogInformation($"C# Kafka trigger function processed a message: {events[i]}");
                var eventDataJson = JObject.Parse(events[i]);
                var message = eventDataJson["Value"].ToString();
                messages[i] = message;
            }
            return messages;
        }
    }
}

File: test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/FunctionApps/java/EventHub/src/main/java/com/function/MultiKafkaTriggerQueueOutput.java
package com.function;

import com.microsoft.azure.functions.annotation.*;

import java.util.*;

import com.microsoft.azure.functions.*;
import org.json.*;

public class MultiKafkaTriggerQueueOutput {
    /**
     * This function consume KafkaEvents on the localhost. Change the topic, EventHubBrokerList, and consumerGroup to fit your enviornment.
     * The function is trigged one for each KafkaEvent
     * @param kafkaEventData
     * @param context
     */
    @FunctionName("MultiKafkaTriggerQueueOutput")
    public void runOne(
            @KafkaTrigger(name = "kafkaTrigger", 
                          topic = "e2e-kafka-java-multi-eventhub", 
                          brokerList="EventHubBrokerList",
                          username = "$ConnectionString",
                          password = "%EventHubConnectionString%",
                          authenticationMode = BrokerAuthenticationMode.PLAIN,
                          protocol = BrokerProtocol.SASLSSL,
                          consumerGroup="$Default",
                          dataType = "string", 
                          cardinality = Cardinality.MANY) List<String> kafkaEvents,
            @QueueOutput(name = "message", 
                         queueName = "e2e-java-multi-eventhub", 
                         connection = "AzureWebJobsStorage") OutputBinding<List<String>> messages,
            final ExecutionContext context) {
        List<String> queueMessages = new ArrayList<String>();
        for (String kafkaEventData : kafkaEvents) {
            context.getLogger().info(kafkaEventData);
            JSONObject kafkaEventJsonObj = new JSONObject(kafkaEventData);
            String message = kafkaEventJsonObj.getString("Value");   
            queueMessages.add(message);         
        }
        messages.setValue(queueMessages);
    }
}
